packages/dicomImageLoader/src/imageLoader/colorSpaceConverters/convertPALETTECOLOR.ts

```diff
@@ -45,51 +45,22 @@ export default function (
 ): void {
   const numPixels = imageFrame.columns * imageFrame.rows;
   const pixelData = imageFrame.pixelData;
-
-  Promise.all([
-    fetchPaletteData(imageFrame, 'red', null),
-    fetchPaletteData(imageFrame, 'green', null),
-    fetchPaletteData(imageFrame, 'blue', null),
-  ]).then(([rData, gData, bData]) => {
-    if (!rData || !gData || !bData) {
-      throw new Error(
-        'The image does not have a complete color palette. R, G, and B palette data are required.'
-      );
-    }
-
-    const len = rData.length;
-    let palIndex = 0;
-    let bufferIndex = 0;
-
-    const start = imageFrame.redPaletteColorLookupTableDescriptor[1];
-    const shift =
-      imageFrame.redPaletteColorLookupTableDescriptor[2] === 8 ? 0 : 8;
-
-    const rDataCleaned = convertLUTto8Bit(rData, shift);
-    const gDataCleaned = convertLUTto8Bit(gData, shift);
-    const bDataCleaned = convertLUTto8Bit(bData, shift);
-
-    if (useRGBA) {
-      for (let i = 0; i < numPixels; ++i) {
-        let value = pixelData[palIndex++];
-
-        if (value < start) {
-          value = 0;
-        } else if (value > start + len - 1) {
-          value = len - 1;
-        } else {
-          value -= start;
-        }
-
-        colorBuffer[bufferIndex++] = rDataCleaned[value];
-        colorBuffer[bufferIndex++] = gDataCleaned[value];
-        colorBuffer[bufferIndex++] = bDataCleaned[value];
-        colorBuffer[bufferIndex++] = 255;
-      }
-
-      return;
-    }
-
+  const rData = imageFrame.redPaletteColorLookupTableData;
```

Comment:
Why not promise all? since OHIF has a metadata provider that create promises, i'm pretty sure this will break OHIF